src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -109,6 +111,13 @@ def test_conversion_as_lone_interface(self):
                 self.case = num
                 self.test_kwargs = kwargs
                 self.interface = self.data_interface_cls(**self.test_kwargs)
+                do_set_probe = isinstance(self.interface, BaseRecordingExtractorInterface)
+                if do_set_probe:
+                    assert isinstance(self.interface, BaseRecordingExtractorInterface)
+                    self.interface.set_probe(
+                        generate_mock_probe(num_channels=self.interface.recording_extractor.get_num_channels()),
+                        group_mode="by_shank",
+                    )
```

Comment:
I would move this section out of the base mixin (which covers things for all data modalities) down into the specific mixing for recording interfaces, then you don't have to have this `isinstance` call

https://github.com/catalystneuro/neuroconv/blob/e2298fa047bace71fc106b65ce4d197767998a0e/src/neuroconv/tools/testing/data_interface_mixins.py#L276

Reply:
But I think it belongs between

```
self.interface = self.data_interface_cls(**self.test_kwargs)
```

and

```
self.check_metadata_schema_valid()
```

so I'm not sure how to move it down into RecordingExtractorInterfaceTestMixin without adding another abstract method such as `self.mutate_the_interface_with_mock_data()`

LMK if I should go that route.

Reply:
You could just override the method entirely on the recording specific child mixin

I also have a suggestion at that level for a better trigger condition to use the mock instead of an existing probe (thinking of SpikeGLX in this case)

```python
class RecordingExtractorInterfaceTestMixin(DataInterfaceTestMixin, TemporalAlignmentMixin):
...

    def test_conversion_as_lone_interface(self):
        interface_kwargs = self.interface_kwargs
        if isinstance(interface_kwargs, dict):
            interface_kwargs = [interface_kwargs]
        for num, kwargs in enumerate(interface_kwargs):
            with self.subTest(str(num)):
                self.case = num
                self.test_kwargs = kwargs
                self.interface = self.data_interface_cls(**self.test_kwargs)

                if self.interface.recording_extractor.has_probe():  # to not override probes that may have been autoset per format
                    assert isinstance(self.interface, BaseRecordingExtractorInterface)
                    self.interface.set_probe(
                        generate_mock_probe(num_channels=self.interface.recording_extractor.get_num_channels()),
                        group_mode="by_shank",
                    )

                self.check_metadata_schema_valid()
                self.check_conversion_options_schema_valid()
                self.check_metadata()
                self.nwbfile_path = str(self.save_directory / f"{self.__class__.__name__}_{num}.nwb")
                self.run_conversion(nwbfile_path=self.nwbfile_path)
                self.check_read_nwb(nwbfile_path=self.nwbfile_path)

                # Any extra custom checks to run
                self.run_custom_checks()
```

Reply:
Do you mean we should override that method and duplicate all of that same code? It's okay with me, but I just wanted to check whether that's what you mean.

Reply:
>  It's okay with me, but I just wanted to check whether that's what you mean.

It is. go ahead